src/server/service/debug.rs

```diff
@@ -546,8 +570,131 @@ where
         self.debugger.reset_to_version(req.get_ts());
         sink.success(ResetToVersionResponse::default());
     }
+
+    fn flashback_to_version(
+        &mut self,
+        ctx: RpcContext<'_>,
+        req: FlashbackToVersionRequest,
+        sink: UnarySink<FlashbackToVersionResponse>,
+    ) {
+        let tmp = self.region_info_accessor.region_leaders();
+        let region_leaders = tmp.read().unwrap();
+
+        // Prepare flashback for specified regions.
+        let start_ts = block_on(self.pd_client.get_tso()).unwrap();
+        let version = req.get_version();
+        let key_range = build_key_range(req.get_start_key(), req.get_end_key(), false);
+        let (prepare_tx, mut prepare_rx) = channel(region_leaders.len());
+
+        let mut flashback_region_num = 0;
+        region_leaders
+            .iter()
+            .filter(|region_id| {
+                req.get_region_ids().is_empty() || req.get_region_ids().contains(region_id)
+            })
+            .filter(|&region_id| {
+                let r = self.debugger.region_info(*region_id).unwrap();
+                let region = r
+                    .region_local_state
+                    .as_ref()
+                    .map(|s| s.get_region().clone())
+                    .unwrap();
+                check_intersect_of_range(
+                    &build_key_range(region.get_start_key(), region.get_end_key(), false),
+                    &key_range.clone(),
+                )
+            })
+            .for_each(|&region_id| {
+                flashback_region_num += 1;
+                let debugger = self.debugger.clone();
+                let prepare_tx_clone = prepare_tx.clone();
+                self.pool.spawn(async move {
+                    match debugger.flashback_to_version(
+                        region_id,
+                        version.clone(),
+                        start_ts.clone(),
+                        TimeStamp::zero(),
+                        prepare_tx_clone,
+                    ) {
+                        Ok(_) => {}
+                        Err(err) => {
+                            return Err(Error::FlashbackFailed(format!(
+                                "prepare flashback failed err is {}",
+                                err
+                            )));
```

Comment:
It seems that the error terminated here. We need to retry these failed regions and ensure that all regions have completed preparation before proceeding to the next step.

Reply:
retry mechanism is implemented in `exec_flashback_region`. 
Only the retry timeout will be terminated here.